tests/testthat/test-enw_add_metaobs_features.R

```diff
@@ -0,0 +1,115 @@
+library(data.table)
+
+# Filter example hospitalisation data to be natioanl and over all ages
+nat_germany_hosp <- setkey(germany_covid19_hosp[
+  (location == "DE") & (age_group %in% "00+")
+], reference_date)
+
+holidays = c(
+  "2021-04-04", "2021-04-05",
+  "2021-05-01", "2021-05-13",
+  "2021-05-24"
+)
+
+junk <- c("Garbage Date")
+
+metadatacols <- c("day_of_week", "day", "week", "month")
+
+test_that("enw_add_metaobs_features datecol arg validated (exists and is.Date)", {
+  expect_error(enw_add_metaobs_features(nat_germany_hosp, datecol = "reference_date"), NA)
+  expect_warning(enw_add_metaobs_features(nat_germany_hosp, datecol = "report_date"))
+  expect_error(enw_add_metaobs_features(nat_germany_hosp))
+  expect_error(enw_add_metaobs_features(nat_germany_hosp, datecol = "location"))
+})
+
+test_that("enw_add_metaobs_features always adds all columns", {
+  expect_equal(
+    sort(intersect(
+      colnames(enw_add_metaobs_features(
+        nat_germany_hosp, datecol = "reference_date"
+      )),
+      metadatacols
+    )),
+    sort(metadatacols)
+  )
+})
+
+test_that("enw_add_metaobs_features overwrites columns with a warning", {
+  dummy <- as.data.table(nat_germany_hosp)
+  dow <- "Placeholder"
+  dummy[, day_of_week := dow ]
+  expect_warning(
+    metaobs <- enw_add_metaobs_features(dummy, datecol = "reference_date")
+  )
+  expect_no_match(
+    as.character(metaobs$day_of_week),
+    dow
+  )
+})
+
+test_that("enw_add_metaobs_features errors when provided unparseable dates.", {
+  expect_error(enw_add_metaobs_features(
+    nat_germany_hosp,
+    datecol = "reference_date",
+    holidays = junk
+  ))
+  expect_error(enw_add_metaobs_features(
+    nat_germany_hosp,
+    datecol = "reference_date",
+    holidays = c(holidays, junk)
+  ))
+})
+
+test_that("enw_add_metaobs_features does not set holidays if `c()` or `NULL` provided", {
+  mobs <- enw_add_metaobs_features(
+    nat_germany_hosp,
+    datecol = "reference_date",
+    holidays = NULL,
+    holidays_to = "Holiday"
+  )
+  expect_equal(mobs[day_of_week == "Holiday", .N], 0)
+  mobs <- enw_add_metaobs_features(
+    nat_germany_hosp,
+    datecol = "reference_date",
+    holidays = c(),
+    holidays_to = "Holiday"
+  )
+  expect_equal(mobs[day_of_week == "Holiday", .N], 0)
+})
+
+test_that("enw_preprocess_data passes arguments to enw_add_metaobs_features", {
+  pobs <- enw_preprocess_data(nat_germany_hosp, holidays = holidays)
```

Comment:
this feels like it should be in `enw_preprocess_data` tests but open to an argument why it shouldn't be.

Reply:
seems reasonable to me, moving over.